VidLoader/VidLoader/Sources/Classes/ResourceLoader/SchemeHandler.swift

```diff
@@ -20,18 +20,23 @@ enum SchemeType: String {
 }
 
 struct SchemeHandler: SchemeHandleable {
-    func urlAsset(with mediaURL: URL?) -> Result<AVURLAsset, ResourceLoadingError> {
+    
+    func urlAsset(with mediaURL: URL?, headers: [String: String]?) -> Result<AVURLAsset, ResourceLoadingError> {
```

Comment:
the `headers: [String: String]?` is unused, maybe you can create the `AVURLAssetHTTPHeaderFieldsKey` cookies array [HTTPCookie] in the scope and remove the usage of `HTTPCookieStorage.shared` saving it in a shared class when we are working with multiple threads can have side effects.

Also one more thing:
If you try to reuse headers, should cookie values also added to the video/playlist request? or it is enough only to create `AVURLAsset` with options?

I mean this place:

PlaylistLoader && ResourceLoader
        headers?.forEach {
            urlRequest.addValue($0.value, forHTTPHeaderField: $0.key)
        }

Reply:
Did you managed to download the video with other way? I tried create and add only to AVURLAsset options with 2 ways below:
    
    `// 1
    func createCookieOptions(headers: [String: String]?, domain: String?) -> [HTTPCookie]? {
        guard let domain = domain else { return nil }
        
        var cookies = [HTTPCookie]()
        if let headers = headers {
            for key in headers.keys {
                let cookie: [HTTPCookiePropertyKey : Any] = [
                    HTTPCookiePropertyKey.domain: domain,
                    HTTPCookiePropertyKey.path: "/",
                    HTTPCookiePropertyKey.secure: true,
                    HTTPCookiePropertyKey.init("HttpsOnly"): true,
                    HTTPCookiePropertyKey.value: headers[key] ?? "",
                    HTTPCookiePropertyKey.name: key,
                ]
                if let httpCookie = HTTPCookie(properties: cookie) {
                    cookies.append(httpCookie)
                }
            }
        }
        return cookies
    }
    
    // 2
    func createCookieOptions(headers: [String: String]?, url: URL) -> [String: [String: String]]? {
        guard let cookie = headers else { return nil }
        var cookies = [HTTPCookie]()
        for key in cookie.keys {
            let cookieField = ["Set-Cookie": "\(key)=\(cookie[key] ?? "")"]
            let cookie = HTTPCookie.cookies(withResponseHeaderFields: cookieField, for: url)
            cookies.append(contentsOf: cookie)
        }
        let values = HTTPCookie.requestHeaderFields(with: cookies)
        return ["AVURLAssetHTTPHeaderFieldsKey": values]
    }
    `

Also adding on urlRequest headers did not work too:
   ` headers?.forEach {
        urlRequest.addValue($0.value, forHTTPHeaderField: $0.key)
      }`
      
I removed the headers parameters from SchemeHandler implementation.